base-notebook/test/test_start_container.py

```diff
@@ -2,56 +2,77 @@
 # Distributed under the terms of the Modified BSD License.
 
 import logging
+from typing import Optional
 import pytest
 import requests
+import re
+import time
 
 from conftest import TrackedContainer
 
 LOGGER = logging.getLogger(__name__)
 
 
 @pytest.mark.parametrize(
-    "env,expected_server",
+    "env,expected_command,expected_start,expected_warnings",
     [
-        (["JUPYTER_ENABLE_LAB=yes"], "lab"),
-        (None, "notebook"),
+        (
+            ["JUPYTER_ENABLE_LAB=yes"],
+            "jupyter lab",
+            True,
+            ["WARNING: JUPYTER_ENABLE_LAB is ignored"],
+        ),
+        (None, "jupyter lab", True, []),
+        (["JUPYTER_CMD=lab"], "jupyter lab", True, []),
+        (["RESTARTABLE=yes"], "run-one-constantly jupyter lab", True, []),
+        (["JUPYTER_CMD=notebook"], "jupyter notebook", True, []),
+        (["JUPYTER_CMD=server"], "jupyter server", True, []),
+        (["JUPYTER_CMD=nbclassic"], "jupyter nbclassic", True, []),
+        (
+            ["JUPYTERHUB_API_TOKEN=my_token"],
+            "jupyterhub-singleuser",
+            False,
+            ["WARNING: using start-singleuser.sh"],
+        ),
     ],
 )
 def test_start_notebook(
```

Comment:
We should probably name this `test_start_jupyter`? :)

Reply:
The name `start-notebook.sh` should also be changed, but let's not touch it in this PR, it will break so many users.

Reply:
Ok @mathbunnyru so maybe I should let it with the same name as the file it tests `test_start_notebook`?